src/NUnitFramework/tests/Assertions/AssertThrowsAsyncTests.cs

```diff
@@ -0,0 +1,214 @@
+﻿using System;
+using System.Threading.Tasks;
+using NUnit.TestUtilities;
+
+namespace NUnit.Framework.Tests.Assertions
+{
+    [TestFixture]
+    public class AssertThrowsAsyncTests
+    {
```

Comment:
Yes, it would. But shared projects aren't supported in all the compilers and ides we support, most notably in MonoDevelop. I'm afraid there are a lot of things we do that would be much easy if we abandoned support for legacy runtimes and older IDEs. We could do that, but then we'd have to call ourselves Microsoft. :-)